### PR TITLE
add query.sql file using Jinja templating to loop through projects

### PR DESCRIPTION
**Pull request by Marlene-M-Hirose**
Query doesn't need to be a Python script. It just needs to be a SQL file. Using Jinja to loop through the projects we want to get data from for bigquery_usage_v2

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-2305)

### CODE REVIEW

**Review by kwindau — APPROVED** [review]

File: sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v2/query.sql [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    error_result.location AS error_location,
+    error_result.reason AS error_reason,
+    error_result.message AS error_message,
+    query_info_resource_warning AS resource_warning,
+  FROM
+    `moz-fx-data-shared-prod.monitoring_derived.jobs_by_organization_v1` AS t1
+  LEFT JOIN
+    UNNEST(referenced_tables) AS referenced_tables
+  ),
+  {#- format off #}
```
Comment (sean-rose) [inline sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v2/query.sql:39]: I know the SQL-with-Jinja code example I showed you used `format off` and `format on` directives to allow manual formatting of the Jinja portion of the query, but in general I'd suggest trying to live with the default formatting and only using manual formatting when the auto-formatting is making the code less readable, as the auto-formatting helps keep our code consistent.

Specifically, in this case I'd suggest:
1. Remove the `format off` and `format on` directives.
2. Run `bqetl format` on this file.
3. If you really don't like how it formatted this section, put the `format off` and `format on` directives back and tweak the formatting as desired.

**Comment by Marlene-M-Hirose** [comment]
> are changes in the DAG needed?
> 
> https://github.com/mozilla/bigquery-etl/blob/ae2edf2234c7137508392f2b2a4318bff53fb2bf/dags/bqetl_monitoring.py#L135

No, it's all automatic now. The DAGs should update automatically here: https://github.com/mozilla/telemetry-airflow-dags

File: sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v2/query.sql (hunk 1 above)
Comment (Marlene-M-Hirose) [reply sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v2/query.sql]: I've removed the `format off` and `format on` directives

File: sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v2/query.sql [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    AS total_terabytes_processed,
+    ROUND(total_bytes_billed / 1024 / 1024 / 1024 / 1024, 4)
+    AS total_terabytes_billed,
+    total_slot_ms,
+    error_result.location AS error_location,
+    error_result.reason AS error_reason,
+    error_result.message AS error_message,
+    query_info_resource_warning AS resource_warning,
+  FROM
+    `moz-fx-data-shared-prod.monitoring_derived.jobs_by_organization_v1` AS t1
```
[the remark this answers is not in the document]
Comment (Marlene-M-Hirose) [reply sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v2/query.sql:35]: I think it originally referred to `tier 1`, but changed it to jobs

File: sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v2/query.sql [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    REGEXP_EXTRACT(query, r'Query ID: (\w+), ') AS query_id,
+  FROM
+    `{{project}}.region-us.INFORMATION_SCHEMA.JOBS_BY_PROJECT` AS jp
+  LEFT JOIN
+    UNNEST(referenced_table) AS referenced_table
+  WHERE
+    DATE(creation_time) = @submission_date {%- if not loop.last %}
+  UNION ALL
+    {%- endif %}
+    {%- endfor %}
```
Comment (sean-rose) [inline sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v2/query.sql]: OK, the auto-formatting of the Jinja is indeed unfortunately kind of janky, especially the `if` statement.  Here's what I'd suggest.
```suggestion
  {#- format off #}
  {%- for project in DEFAULT_PROJECTS %}
    SELECT
      jp.project_id AS source_project,
      DATE(creation_time) AS creation_date,
      job_id,
      referenced_table.project_id AS reference_project_id,
      referenced_table.dataset_id AS reference_dataset_id,
      referenced_table.table_id AS reference_table_id,
      user_email,
      REGEXP_EXTRACT(query, r'Username: (.*?),') AS username,
      REGEXP_EXTRACT(query, r'Query ID: (\w+), ') AS query_id,
    FROM
      `{{project}}.region-us.INFORMATION_SCHEMA.JOBS_BY_PROJECT` AS jp
    LEFT JOIN
      UNNEST(referenced_table) AS referenced_table
    WHERE
      DATE(creation_time) = @submission_date
    {%- if not loop.last %}
      UNION ALL
    {%- endif %}
  {%- endfor %}
  {#- format on #}
```

**Pull request merged**